homeassistant/components/bmw_connected_drive/config_flow.py

```diff
@@ -101,24 +100,25 @@ async def async_step_user(
                 errors["base"] = "invalid_auth"
 
             if info:
-                if self.source == SOURCE_REAUTH:
+                if self._existing_config_entry:
```

Comment:
```python
                if self.source == SOURCE_REAUTH:
                    return self.async_update_reload_and_abort(
                        self._get_reauth_entry(),...)
                if self.source == SOURCE_RECONFIGURE:
                    return self.async_update_reload_and_abort(
                        self._get_reconfigure_entry(),...)
```

Reply:
Just a hint - on reconfigure, the `reason` must be explicitely stated.

Not sure, but if this will be done over all integrations, maybe it makes sense to use `self.init_step` or `self.source` as default for `async_update_reload_and_abort`?

Reply:
Is this what you had in mind? #127756

Reply:
https://github.com/home-assistant/core/pull/127756 has been merged, so the reason can be dropped (but still needs to be added to `strings.json`)

Reply:
Yes, thank you! Will adjust this evening.